test/junit/scala/PredefTest.scala

```diff
@@ -0,0 +1,16 @@
+package scala
+
+import org.junit.Assert._
+import org.junit.Test
+
+
+class PredefTest {
+  @Test
+  def testTuple2Alias: Unit = {
+    val tup = "foobar" -> 3
+    val char = tup match {
+      case str -> i => str.charAt(i)
```

Comment:
Just realized a small nit: should we test nested tuples? Just to be sure both arrows have the same associativity.